### ISSUE TITLE
"Library not found" error for SyclesPlugin.dll upon Addon Install (Softimage 2015 SP2 64-bit)

### ISSUE THREAD

**Issue by PsifinityDXI** [Tugcga/Sycles#19]
![2023-10-17_16-43](https://github.com/Tugcga/Sycles/assets/20031620/d966811e-ebe4-4a1e-85db-d18aca7581a5)

![2023-10-17_16-55](https://github.com/Tugcga/Sycles/assets/20031620/30e1b6e0-09fa-4f2f-ba3b-ac3801b1c998)

Sycles 2.13 doesn't seem to work for at least my laptop, as Sycles 1.12 installs and works just fine.

It's important to mention that my laptop doesn't support/open Blender 3.5 and up due to an outdated graphics driver (Intel HD Graphics 4000, can't upgrade it either)—so if the .dll is attempting to call or refer to something that isn't there and fails, then that sounds like the reason why Softimage declares the plugin as "invalid" (correct me if I'm wrong though, kinda just taking a guess).

Blender 3.4.1 and below do work though, if that means anything. I also noted that Sycles 1.12 doesn't require that .dll at all to work, as it is not present within the nt-x86-64 folder. I would definitely like your input on this, if possible.

**Comment by Tugcga** [comment]
You can try to find the missing dll by using ```Dependency Walker``` application. Also try to download ```nvcuda.dll``` from [here](https://github.com/Tugcga/Sycles/releases/download/binaries.2.1/nvcuda.dll). It's for CUDA, but it may required even for systems without nvidia cards.

**Comment by PsifinityDXI** [comment]
> You can try to find the missing dll by using `Dependency Walker` application. Also try to download `nvcuda.dll` from [here](https://github.com/Tugcga/Sycles/releases/download/binaries.2.1/nvcuda.dll). It's for CUDA, but it may required even for systems without nvidia cards.

Solved the issue!

I downloaded nvcuda.dll to C:\Windows\System32 at first, and Softimage cleared up the "invalid" status.
I just saw the main page for Sycles that says to place it in the "bin" folder of the plugin though, so I moved it there and it still works fine!!

Only issue I'm running into now is HDRMaster deciding to not work with Sycles 2.1.3 for some reason—not sure if it's because Sycles 2.1.3 is still in WIP, or it's just that the HDRM plugin itself needs to be updated lol.

![2023-10-18_23-59](https://github.com/Tugcga/Sycles/assets/20031620/413e42d4-6605-4872-bb4b-951713b7f22c)

**Comment by Tugcga** [comment]
HDRMaster is tool that create environment light source for any selected hdr image and tweak parameters of this light source. It can works only with Sycles because it create environment light only from Sycles. Arnold or MentalRay can not render these lights. Potentially it should work with any version of the Sysles (1.x or 2.x). There is only one (but important) limitation: for 2.x it can not tweak parameters of the created environment light, because there are some changes of the node setups with respect to 1.x. It's easy to fix, but I can not find the time for this action.

**Comment by PsifinityDXI** [comment]
Yeah I can confirm it works with Sycles 1.12—just that attempting to create the primitive with Sycles 2.x just gives that error message, for now anyway.

Either way, thanks for your help with the library issue! Turns out that I really should've just read the front page and checked my computer LMAO.
